docs/VLSI/Tutorial.rst

```diff
@@ -0,0 +1,138 @@
+
+* example-vlsi
+
+  * Entry point to Hammer. Contains example placeholders for hooks.
+
+* example.v
+
+  * Verilog wrapper around the accelerator and dummy hard macro.
+
+* example.yml
+
+  * Hammer IR for this tutorial.
+
+* extra_libraries
+
+  * Contains collateral for the dummy hard macro.
+
+* generated-src
+
+  * All of the elaborated Chisel and FIRRTL.
+
+* hammer, hammer-<vendor>-plugins, hammer-<tech>-plugin
+
+  * Core, tool, tech repositories.
+
+Prerequisites
+-------------
+
+* Python 3.4+
+* numpy and gdspy packages
+* Genus, Innovus, and Calibre licenses
+* For ASAP7 specifically:
+
+  * Download the `ASAP7 PDK <http://asap.asu.edu/asap/>`__ tarball and do not extract it
+  * If you have additional ASAP7 hard macros, their LEF & GDS need to be 4x upscaled @ 4000 DBU precision
+
+Initial Setup
+-------------
+In the Chipyard root, run:
+
+.. code-block:: shell
+
+    ``./scripts/init-vlsi.sh asap7`` 
+    
+to pull the HAMMER & plugin submodules. Note that for technologies other than ``asap7``, the tech submodule must be added in the ``vlsi`` folder first.
+
+Pull the Hammer environment into the shell:
+
+.. code-block:: shell
+
+    cd vlsi
+    export HAMMER_HOME=$PWD/hammer
+    source $HAMMER_HOME/sourceme.sh
+
+Building the Design
+-------------------
+To elaborate the Sha3RocketConfig (Rocketchip w/ the accelerator) and set up all prerequisites for the build system to push just the accelerator + hard macro through the flow:
+
+.. code-block:: shell
+
+    make buildfile MACROCOMPILER_MODE='--mode synflops' CONFIG=Sha3RocketConfig VLSI_TOP=Sha3AccelwBB
+
+The ``MACROCOMPILER_MODE='--mode synflops'`` is needed because the ASAP7 process does not yet have a memory compiler. Therefore, flip-flop arrays are used instead.
+
+The ``CONFIG=Sha3RocketConfig`` selects the target generator config in the same manner as the rest of the Chipyard framework. This elaborates a Rocket-chip with the Sha3Accel module.
+
+The ``VLSI_TOP=Sha3AccelwBB`` indicates that we are only interested in physical design of the accelerator block. If this variable is not set, the entire SoC will be pushed through physical design. Note that you should not set the ``TOP`` variable because it is used during Chisel elaboration.
+
+For the curious, ``make buildfile`` generates a set of Make targets in ``build/hammer.d``. It needs to be re-run if environment variables are changed. It is recommended that you edit these variables directly in the Makefile rather than exporting them to your shell environment.
+
+Running the VLSI Flow
+---------------------
+
+example-vlsi
+^^^^^^^^^^^^
+This is the entry script with placeholders for hooks. In the ``ExampleDriver`` class, a list of hooks is passed in the ``get_extra_par_hooks``. Hooks are additional snippets of python and TCL (via ``x.append()``) to extend the Hammer APIs. Hooks can be inserted using the ``make_pre/post/replacement_hook`` methods.
```

Comment:
Just to make sure... is "pre/post/replacement" described somewhere? If not, you might want a quick description. Something like "where these methods inject the statements before, after or completely replace the hook"